### ISSUE TITLE
used_underscore_items should not warn about uses of foreign functions

### ISSUE THREAD

**Issue by taiki-e** [rust-lang/rust-clippy#14156]
### Summary

Currently, this lint warns about uses of foreign functions (functions in an extern block), but since the names of these functions are defined by the C (or other) library that exports them, there is no way to deal with them on our code side other than to allow lint or to use the redundant link_name attribute.

### Lint Name

used_underscore_items

### Reproducer

I tried this code:

```rust
#![warn(clippy::used_underscore_items)]

extern "C" {
    pub fn _exit(code: i32) -> !;
}

fn _f() {
    unsafe { _exit(1) }
}
```

I saw this happen:

```
warning: used underscore-prefixed item
 --> src/lib.rs:8:14
  |
8 |     unsafe { _exit(1) }
  |              ^^^^^^^^
  |
note: item is defined here
 --> src/lib.rs:4:5
  |
4 |     pub fn _exit(code: i32) -> !;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#used_underscore_items
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![warn(clippy::used_underscore_items)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I expected to see this happen: no warning

[playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=c1d5303d4791318543e40276ab39a744)

### Version

```text
rustc 1.86.0-nightly (bef3c3b01 2025-02-04)
binary: rustc
commit-hash: bef3c3b01f690de16738b1c9f36470fbfc6ac623
commit-date: 2025-02-04
host: aarch64-apple-darwin
release: 1.86.0-nightly
LLVM version: 19.1.7
```

**Comment by taiki-e** [comment]
cc @WeiTheShinobi (who implemented this lint in https://github.com/rust-lang/rust-clippy/pull/13294)
